services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/HankeControllerITests.kt

```diff
@@ -329,14 +330,19 @@ class HankeControllerITests(@Autowired override val mockMvc: MockMvc) : Controll
 
         @Test
         fun `Sanitize hanke input and return 200`() {
-            val hanke = HankeFactory.create().apply { generated = true }
-            every { hankeService.createHanke(hanke.copy(id = null, generated = false)) } returns
-                hanke.copy(generated = false)
+            val hanke = HankeFactory.create().withYhteystiedot().apply { generated = true }
+            val expectedServiceArgument =
+                hanke.apply {
+                    generated = false
+                    id = null
+                }
```

Comment:
Apply changes the object, doesn't create a new one.
I.e. these asserts fail:
```kotlin
            val hanke = HankeFactory.create().withYhteystiedot().apply { generated = true }
            val expectedServiceArgument =
                hanke.apply {
                    generated = false
                    id = null
                }
            assertThat(hanke.generated).isTrue()
            assertThat(hanke.id).isNotNull()
```
So this test is now calling the API with a sanitized hanke. It's safer to create a new hanke:

```suggestion
            val expectedServiceArgument =
                HankeFactory.create(id = null).withYhteystiedot().apply { generated = false }
```

Reply:
Thanks, fixed.